jobs: TestRetriesWithExponentialBackoff failed

jobs.TestRetriesWithExponentialBackoff [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592196?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592196?buildTab=artifacts#/) on master @ [591eec9f33a36d0749ec87c6c3141dd275eae81f](https://github.com/cockroachdb/cockroach/commits/591eec9f33a36d0749ec87c6c3141dd275eae81f):


```
=== RUN   TestRetriesWithExponentialBackoff
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0bb61de08ed0e92dbe71ab2af75e6d1f/logTestRetriesWithExponentialBackoff1170518195
    test_log_scope.go:81: use -show-logs to present logs inline
    registry_test.go:1046: -- test log scope end --
--- FAIL: TestRetriesWithExponentialBackoff (34.81s)
=== RUN   TestRetriesWithExponentialBackoff/revert_on_cancel
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    registry_test.go:878: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:878
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:1003
        	Error:      	Not equal: 
        	            	expected: 19
        	            	actual  : 20
        	Test:       	TestRetriesWithExponentialBackoff/revert_on_cancel
        	Messages:   	unexpected number of jobs resumed in retry (post waitFn) 16
    --- FAIL: TestRetriesWithExponentialBackoff/revert_on_cancel (3.65s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/jobs @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetriesWithExponentialBackoff.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>


Jira issue: CRDB-39455

jobs.TestRetriesWithExponentialBackoff [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592268?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592268?buildTab=artifacts#/) on master @ [ffb046601a007042e7768055a952e771f62990e6](https://github.com/cockroachdb/cockroach/commits/ffb046601a007042e7768055a952e771f62990e6):


```
=== RUN   TestRetriesWithExponentialBackoff
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0bb61de08ed0e92dbe71ab2af75e6d1f/logTestRetriesWithExponentialBackoff781452669
    test_log_scope.go:81: use -show-logs to present logs inline
    registry_test.go:1046: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0bb61de08ed0e92dbe71ab2af75e6d1f/logTestRetriesWithExponentialBackoff781452669
--- FAIL: TestRetriesWithExponentialBackoff (26.86s)
=== RUN   TestRetriesWithExponentialBackoff/revert_on_cancel
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    registry_test.go:878: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:878
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:1003
        	Error:      	Not equal: 
        	            	expected: 47
        	            	actual  : 48
        	Test:       	TestRetriesWithExponentialBackoff/revert_on_cancel
        	Messages:   	unexpected number of jobs resumed in retry (post waitFn) 44
    --- FAIL: TestRetriesWithExponentialBackoff/revert_on_cancel (4.86s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetriesWithExponentialBackoff.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

jobs.TestRetriesWithExponentialBackoff [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592978?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/15592978?buildTab=artifacts#/) on master @ [e09f5356381704e404eba1d771152f73c7c8a572](https://github.com/cockroachdb/cockroach/commits/e09f5356381704e404eba1d771152f73c7c8a572):


```
=== RUN   TestRetriesWithExponentialBackoff
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/0bb61de08ed0e92dbe71ab2af75e6d1f/logTestRetriesWithExponentialBackoff594382170
    test_log_scope.go:81: use -show-logs to present logs inline
    registry_test.go:1046: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/0bb61de08ed0e92dbe71ab2af75e6d1f/logTestRetriesWithExponentialBackoff594382170
--- FAIL: TestRetriesWithExponentialBackoff (16.09s)
=== RUN   TestRetriesWithExponentialBackoff/revert_on_cancel
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    registry_test.go:878: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:878
        	            				github.com/cockroachdb/cockroach/pkg/jobs/registry_test.go:1003
        	Error:      	Not equal: 
        	            	expected: 30
        	            	actual  : 31
        	Test:       	TestRetriesWithExponentialBackoff/revert_on_cancel
        	Messages:   	unexpected number of jobs resumed in retry (post waitFn) 27
    --- FAIL: TestRetriesWithExponentialBackoff/revert_on_cancel (1.47s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRetriesWithExponentialBackoff.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/jobs/adopt.go
+++ b/pkg/jobs/adopt.go
@@ -481,7 +481,6 @@ func (r *Registry) maybeClearLease(job *Job, jobErr error) {
 	if jobErr == nil {
 		return
 	}
-	r.unregister(job.ID())
 	r.clearLeaseForJobID(job.ID(), r.db.Executor(), nil /* txn */)
 }
 

--- a/pkg/jobs/registry_test.go
+++ b/pkg/jobs/registry_test.go
@@ -637,8 +637,6 @@ func TestRetriesWithExponentialBackoff(t *testing.T) {
 	defer leaktest.AfterTest(t)()
 	defer log.Scope(t).Close(t)
 
-	skip.WithIssue(t, 125433)
-
 	const (
 		// Number of retries should be reasonably large such that they are sufficient
 		// to test long delays but not too many to increase the test time.

